pkg/security/module/server.go

```diff
@@ -323,7 +323,7 @@ func (a *APIServer) RunSelfTest(ctx context.Context, params *api.RunSelfTestPara
 		}, nil
 	}
 
-	if err := a.module.selfTester.RunSelfTest(); err != nil {
+	if err := a.module.selfTester.RunSelfTest(a.module); err != nil {
```

Comment:
Not a huge fan of `a.Do(a)`, is there no better way ?

Reply:
Indeed. I've make some modifications, it should be better, tell me if it's ok for you